pkg/kubelet/cm/deviceplugin/device_plugin_stub.go

```diff
@@ -144,8 +157,12 @@ func (m *Stub) Update(devs []*pluginapi.Device) {
 // Allocate does a mock allocation
 func (m *Stub) Allocate(ctx context.Context, r *pluginapi.AllocateRequest) (*pluginapi.AllocateResponse, error) {
 	log.Printf("Allocate, %+v", r)
+	if m.allocFunc != nil {
+		return m.allocFunc(r, m.devs)
```

Comment:
Avoid nil checks via default no-op implementations.

Reply:
Sorry i'm not very clear about this comment, do you mean just `return m.allocFunc(r, m.devs)` without if?

Reply:
Yep and in the constructor if allocFun is nil set it to a no-op function

Reply:
OK, thanks, add `defaultAllocFunc()` as no-op function.